homeassistant/components/slide_local/config_flow.py

```diff
@@ -0,0 +1,152 @@
+        except (AuthenticationFailed, DigestAuthCalcError):
+            return {"base": "invalid_auth"}
+        except Exception:  # noqa: BLE001
+            _LOGGER.exception("Exception occurred during connection test")
+            return {"base": "unknown"}
+
+        return {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the user step."""
+        errors = {}
+        if user_input is not None:
+            self._async_abort_entries_match({CONF_HOST: user_input[CONF_HOST]})
+            user_input[CONF_API_VERSION] = int(user_input[CONF_API_VERSION])
+
+            if not (errors := await self.async_test_connection(user_input)):
+                await self.async_set_unique_id(self._mac)
+                user_input |= {CONF_MAC: format_mac(self._mac)}
+
+                return self.async_create_entry(
+                    title=user_input[CONF_HOST], data=user_input
+                )
+
+        if user_input is not None and user_input.get(CONF_HOST) is not None:
+            self._host = user_input[CONF_HOST]
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST, default=self._host): str,
+                    vol.Required(CONF_PASSWORD): str,
+                    vol.Required(CONF_API_VERSION, default="2"): API_VERSION_SELECTOR,
+                    vol.Required(CONF_INVERT_POSITION, default=False): BOOLEAN_SELECTOR,
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_zeroconf(
+        self, discovery_info: ZeroconfServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle zeroconf discovery."""
+
+        # id is in the format 'slide_000000000000'
+        self._mac = format_mac(str(discovery_info.properties.get("id"))[6:])
+
+        await self.async_set_unique_id(self._mac)
+
+        self._abort_if_unique_id_configured({CONF_HOST: discovery_info.host})
+
+        self._host = discovery_info.host
+
+        return await self.async_step_zeroconf_confirm()
+
+    async def async_step_zeroconf_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the zeroconf_confirm step."""
+        errors = {}
+        if user_input is not None:
+            user_input |= {CONF_HOST: self._host}
+            user_input |= {CONF_API_VERSION: 2}
+            user_input |= {CONF_MAC: format_mac(self._mac)}
+
+            if not (errors := await self.async_test_connection(user_input)):
+                return self.async_create_entry(
+                    title=user_input[CONF_HOST], data=user_input
+                )
+
+        return self.async_show_form(
+            step_id="zeroconf_confirm",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_INVERT_POSITION, default=False): BOOLEAN_SELECTOR,
```

Comment:
I think OptionsFlow is fine for me to change on runtime. Would it also be possible (or rather allowed) to save it into options in ConfigFlow, so that it can be changed in OptionsFlow later on? It seems more user friendly to me if you can already select the correct option during setup.

Reply:
I think it's weird to add the device and then just be asked if you want to invert it without knowing what the current behaviour is.

So with an options flow you first have a little time to check how it works and if it works like you want to use it, otherwise you invert it using the options flow.

This is the practical reason, the other reason is that we want config flow to be about everything needed for connecting and options for configuration

Reply:
So we would setup the device without need of additional information. Do you have a good example for this (I think we still need a confirmation dialog and let the user to press ok).

Reply:
I think we do this with AirGradient